### PR TITLE
Arch: removing unnecessary 'if' conditional check

### PR DESCRIPTION
**Pull request by ebrahimraeyat**

### CODE REVIEW

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7368">pipeline status</a> for feature branch [PR_7368](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7368). Pipeline [613390211](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/613390211) was triggered at [c3e724d](https://github.com/FreeCAD/FreeCAD/pull/7368/commits/c3e724d78dce89459e5d446748ac4a602f372e04). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/Arch/ArchRebar.py [hunk 1]
```diff
@@ -468,28 +468,18 @@ def execute(self,obj):
...
-                shapes.append(bar)
-                w = basewire.copy()
-                w.Placement = obj.PlacementList[i]
-                self.wires.append(w)
+            bar = bar.copy()
+            bar.Placement = obj.PlacementList[i]
+            shapes.append(bar)
+            w = basewire.copy()
+            w.Placement = obj.PlacementList[i]
+            self.wires.append(w)
```
Comment (benj5378) [inline src/Mod/Arch/ArchRebar.py:482]: Why can the specific logic for `i == 0` be removed?

Comment (ebrahimraeyat) [reply src/Mod/Arch/ArchRebar.py:482]: > Why can the specific logic for `i == 0` be removed?

bar is a Part shape. we can simply copy it for each bar that must be add. The first one is not something special.

**Comment by luzpaz** [comment]
@ebrahimraeyat please amend commit 

`git commit --amend -m "Arch: removing unnecessary 'if' conditional check" -m "Fix for ArchRebar.py"`

**Comment by luzpaz** [comment]
CC @amrit3701 (rebar wb dev)

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7368">pipeline status</a> for feature branch [PR_7368](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7368). Pipeline [620950341](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/620950341) was triggered at [216ab71](https://github.com/FreeCAD/FreeCAD/pull/7368/commits/216ab71ade1db765cac31f70eb3581fafaaa878e). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/Arch/ArchRebar.py (hunk 1 above)
Comment (amrit3701) [reply src/Mod/Arch/ArchRebar.py:482]:
```suggestion
           if i != 0:
               bar = bar.copy()
               basewire = basewire.copy()
            bar.Placement = obj.PlacementList[i]
            shapes.append(bar)
            basewire.Placement = obj.PlacementList[i]
            self.wires.append(basewire)
```

I think we can use above logic. @ebrahimraeyat with your logic you are creating new `bar` and `basewire` objects  for `i=0` but objects are already there. Yes, Python has good garbage collector but here we can control object creation.

**Review by amrit3701 — CHANGES_REQUESTED** [review]

**Comment by ebrahimraeyat** [comment]
I didn't check the performance, but for creating stirrups we must check i value each time we create an stirrup, while bar and basewire are Part shape and not object and I think this is not important to create 1 extra copy of bar shape.

**Review by amrit3701 — APPROVED** [review]
LGTM!

**Comment by yorikvanhavre** [comment]
I agree the extra object creation is not really important... shape.copy() is fast enough in this case (IIRC @realthunder discovered an even faster way but I can't remember) and the extra object should indeed be garbage-collected afterwards

**Pull request merged**